src/core/evaluator.js

```diff
@@ -1640,19 +1650,22 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
                 xObjStateManager.transform(matrix);
               }
 
+              // enqueue textContent chunk before call for sub-form.
+              enqueueChunk();
               next(self.getTextContent({
                 stream: xobj,
                 task,
                 resources: xobj.dict.get('Resources') || resources,
                 stateManager: xObjStateManager,
                 normalizeWhitespace,
                 combineTextItems,
+                sink
               }).then(function (formTextContent) {
-                Util.appendToArray(textContent.items, formTextContent.items);
-                Util.extendObj(textContent.styles, formTextContent.styles);
+                // Util.appendToArray(textContent.items, formTextContent.items);
+                // Util.extendObj(textContent.styles, formTextContent.styles);
 
-                xobjsCache.key = name;
-                xobjsCache.texts = formTextContent;
+                // xobjsCache.key = name;
+                // xobjsCache.texts = formTextContent;
```

Comment:
I've looked around quickly, but have thus far not found a case where caching the `textContent` helps.
*However*, that doesn't mean that the caching is without value; please see below!

For documents with a *huge* number of `XObject`s (such as #8380), despite *non* of its `XObject`s being of the `/Type /Form` kind, the cache still helps a lot.
The explanation is that [evaluator.js#L1608](https://github.com/mozilla/pdf.js/blob/593dec1bb7aec1802abf8268137b0f7adab2ae32/src/core/evaluator.js#L1608) means that we can bail out in [evaluator.js#L1613](https://github.com/mozilla/pdf.js/blob/593dec1bb7aec1802abf8268137b0f7adab2ae32/src/core/evaluator.js#L1613), before having to do a bunch of redundant `Dict` lookups in [evaluator.js#L1616-L1623](https://github.com/mozilla/pdf.js/blob/593dec1bb7aec1802abf8268137b0f7adab2ae32/src/core/evaluator.js#L1616-L1623).

I've tried disabling the caching, for the PDF file in #8380, and the run-time of `getTextContent` *tripled* (from ~200ms to ~600ms).

So, to try and summarize: I'd agree that we might not need to cache previous `textContent` as we currently do. However, I'm thinking that we should still cache (the names of) non-`Form` XObjects so that we can avoid unnecessary repeated `Dict` lookups.

---
Sorry, the above became a bit long; but I hope it makes sense!